language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/imports/OptimizeImportsCodeAction.java

```diff
@@ -73,10 +73,10 @@ public List<CodeAction> getCodeActions(CodeActionContext context,
         List<ImportDeclarationNode> fileImports = new ArrayList<>();
         ((ModulePartNode) syntaxTree.rootNode()).imports().stream().forEach(fileImports::add);
 
-        List<LineRange> toBeRemovedImportsLocations = context.diagnostics(context.filePath()).stream()
+        List<LineRange> toBeRemovedImportsLocations = new ArrayList<>(context.diagnostics(context.filePath()).stream()
```

Comment:
Let's leave this as is as well, unless we can factor the forloop below to avoid having to mutate this list

Reply:
I left it as it was. Fixed in [2778660](https://github.com/ballerina-platform/ballerina-lang/pull/42904/commits/277866060d74b08115089625bb7d582c095697a1)